File: cli/box.py
from pysha import *
import sys

try:
    command = sys.argv[1]
except:
    print("""You Need To Insert Command.
Usage : 
    box.py command [arguments]

Commands : 
    install    ->   To Install Extention. It Needs One Argument Containing Extention Name.
    uninstall  ->   To Remove Extention. It Needs One Argument Containing Extention Name.
    verify     ->   It Will Verify All Of Your Extentions And Warns You On Unverified Extentions.
    list       ->   Gets Extention List For You.
    help       ->   Show This Help Message.
""")
    exit(1)


def install_command(name):
    pass


def uninstall_command(name):
    pass


def verify_command():
    pass


def list_command():
    pass


def help_command():
    print("""You Need To Insert Command.
Usage : 
    box.py command [arguments]

Commands : 
    install    ->   To Install Extention. It Needs One Argument Containing Extention Name.
    uninstall  ->   To Remove Extention. It Needs One Argument Containing Extention Name.
    verify     ->   It Will Verify All Of Your Extentions And Warns You On Unverified Extentions.
    list       ->   Gets Extention List For You.
    help       ->   Show This Help Message.
""")


try:
    Switch(command).cases({
        Case("install"):
            [install_command, [sys.argv[2]]],
        Case("uninstall"):
            [uninstall_command, [sys.argv[2]]],
        Case("verify"):
            [verify_command],
        Case("list"):
            [list_command],
        Case("help"):
            [help_command],
    })
except:
    print("You Need To Insert Argument For Your Command Too ( Extention Name ).")
    exit(1)

File: cli/box.rb
help_string = <<END_HELP
You Need To Insert Command.
Usage :
	box.rb command [arguments]

Commands :
	install    ->   To Install Extention. It Needs One Argument Containing Extention Name.
	uninstall  ->   To Remove Extention. It Needs One Argument Containing Extention Name.
	verify     ->   It Will Verify All Of Your Extentions And Warns You On Unverified Extentions.
	list       ->   Gets Extention List For You.
	search     ->   Search For Your Extention. Needs One Argument As Extention Name.
	help       ->   Show This Help Message.
END_HELP



command = ARGV[0]
if not command
	puts help_string
	exit(1)
end


def help_command
	puts help_string
end

def install(name)
	if name
		puts "Installing " + name
	else
		puts "You Need To Insert Argument For Your Command Too ( Extention Name )."
	end
end

def uninstall(name)
	if name
		puts "Uninstalling " + name
	else
		puts "You Need To Insert Argument For Your Command Too ( Extention Name )."
	end
end

def verify(name)
	if name
		puts "Verifing " + name
	else
		puts "You Need To Insert Argument For Your Command Too ( Extention Name )."
	end
end

def list
	puts "Extention List"
end

def search(name)
	if name
		puts "Searching " + name
	else
		puts "You Need To Insert Argument For Your Command Too ( Extention Name )."
	end
end

case command
when "install"
	install(ARGV[1])
when "uninstall"
	uninstall(ARGV[1])
when "verify"
	verify(ARGV[1])
when "list"
	list
when "help"
	help_command
when "search"
	search(ARGV[1])
else
	puts "There Is No Command " + command
	exit(2)
end
